packages/time-picker/test/value-commit.test.js

```diff
@@ -0,0 +1,390 @@
+        expectValueCommit('12:00');
+      });
+
+      it('should commit on close with outside click', () => {
+        outsideClick();
+        expectValueCommit('12:00');
+      });
+
+      it('should revert and validate on close with Escape', async () => {
+        await sendKeys({ press: 'Escape' });
+        await sendKeys({ press: 'Escape' });
+        expectValidationOnly();
+        expect(timePicker.inputElement.value).to.equal(initialInputElementValue);
+      });
+    });
+
+    describe('entering unparsable input', () => {
+      beforeEach(async () => {
+        timePicker.inputElement.select();
+        await sendKeys({ type: 'INVALID' });
+      });
+
+      it('should commit an empty value on blur', () => {
+        timePicker.blur();
+        expectValueCommit('');
+        expect(timePicker.inputElement.value).to.equal('INVALID');
+      });
+
+      it('should commit an empty value on Enter', async () => {
+        await sendKeys({ press: 'Enter' });
+        expectValueCommit('');
+        expect(timePicker.inputElement.value).to.equal('INVALID');
+      });
+
+      it('should commit an empty value on close with outside click', () => {
+        outsideClick();
+        expectValueCommit('');
+        expect(timePicker.inputElement.value).to.equal('INVALID');
+      });
+
+      it('should revert and validate on close with Escape', async () => {
+        await sendKeys({ press: 'Escape' });
+        expectValidationOnly();
+        expect(timePicker.inputElement.value).to.equal(initialInputElementValue);
+      });
+    });
+
+    describe('clearing input with Backspace', () => {
+      beforeEach(async () => {
+        timePicker.inputElement.select();
+        await sendKeys({ press: 'Backspace' });
+      });
+
+      it('should commit on blur', () => {
+        timePicker.blur();
+        expectValueCommit('');
+      });
+
+      it('should commit on Enter', async () => {
+        await sendKeys({ press: 'Enter' });
+        expectValueCommit('');
+      });
+
+      it('should commit on outside click', () => {
+        outsideClick();
+        expectValueCommit('');
+      });
+
+      it('should revert and validate on close with Escape', async () => {
+        await sendKeys({ press: 'Escape' });
+        expectValidationOnly();
+        expect(timePicker.inputElement.value).to.equal(initialInputElementValue);
+      });
+    });
+
+    describe('with clear button', () => {
+      beforeEach(() => {
+        timePicker.clearButtonVisible = true;
+      });
+
+      it('should clear on clear button click', () => {
+        timePicker.$.clearButton.click();
+        expectValueCommit('');
+      });
+
+      it('should clear on Escape', async () => {
+        await sendKeys({ press: 'Escape' });
+        expectValueCommit('');
+      });
+    });
+  });
+
+  describe('with step', () => {
+    beforeEach(() => {
+      timePicker.step = 1;
+    });
+
+    it('should commit on ArrowUp', async () => {
```

Comment:
This test fails due to multiple value-changed events: https://github.com/vaadin/web-components/pull/6432